packages/api/src/repositories/egg.ts

```diff
@@ -99,17 +99,22 @@ export class EggRepository {
     return ((await this.collection.findOne({ key })) as Egg) || null
   }
 
-  public async list(): Promise<Array<Egg>> {
+  public async list(): Promise<Array<IndexedEgg>> {
     // Get only claimed eggs
     const eggs = await this.collection.find({
       token: { $exists: true },
     })
 
-    return (await eggs.toArray()).map((egg) => ({
+    const eggsArray: Array<Egg> = (await eggs.toArray()) as Array<Egg>
+
+    const eggsByRarity = this.getEggsByRarity(eggsArray)
```

Comment:
Is it possible to modify the line 104, so that the query sorts by `score` and then we "enumerate" the result?